runtime/pkg/graceful/http.go

```diff
@@ -33,29 +36,30 @@ func ServeHTTP(ctx context.Context, server *http.Server, options ServeOptions) e
 		return err
 	}
 
-	// Channel to signal server has stopped
-	serveErrCh := make(chan error)
-	// Start server in a goroutine
+	cctx, cancel := context.WithCancel(ctx)
+	var serveErr error
```

Comment:
Doesn't this code re-introduce the race condition fixed here? https://github.com/rilldata/rill/pull/6930

Bad merge?